File: Java/String-2/src/xyzThere/XyzThere.java
package xyzThere;

public class XyzThere {
    //Retorne true se na String dada houver "xyz", porém não contará os casos em que ele for precedido por um ponto ".".

    public boolean xyzThere(String str) {
        for (int i = 0; i < str.length()-2; i++) {
            if (str.startsWith("xyz",i) && !str.startsWith(".",i-1)){
                return true;
            }
        }
        return false;
    }
}

File: Python/String-2/xyz_there.py
"""
Dada uma String "str", retorne true se ela possuir a string "xyz", contanto que
ela não seja precedida por um ponto ".".
"""


def xyz_there(str):
    for i in range(len(str)):
        if str[i:i+3] == "xyz" and not str[i-1:i] == ".":
            return True
    return False


print(xyz_there('abc.xyz'))
